tests/conftest.py

```diff
@@ -129,28 +131,88 @@ def _test_images_s3_bucket():
 
         self._root_path = defs.TEST_RESULTS_DIR
+        self._results_file = None
+        self._baselines_file = None
 
         # Create the root directory, if it doesn't exist.
         self._root_path.mkdir(exist_ok=True)
 
-        self._file = open(self._root_path / test_name, flags)
-
-    def writeln(self, data: str):
-        """Write the `data` string to the output file, appending a newline."""
-        self._file.write(data)
-        self._file.write("\n")
-        self._file.flush()
+        if dump_results:
+            self._results_file = os.path.join(
+                self._root_path, "{}_results_{}.json".format(
+                    test_name, utils.get_host_kernel_version()))
+        if generate_baselines:
+            self._baselines_file = os.path.join(
+                self._root_path, "{}_config_{}.json".format(
+                    test_name, utils.get_host_kernel_version()))
+            open(self._baselines_file, "w+")
+            shutil.copy(baselines_file, self._baselines_file)
+
+    @staticmethod
+    def __dump_pretty_json(file, data, flags):
+        """Write the `data` dictionary to the output file in pretty format."""
+        with open(file, flags) as file_fd:
+            json.dump(data, file_fd, indent=4)
+            file_fd.write("\n")  # Add newline cause Py JSON does not
+
+    # pylint: disable=R1702
+    def __gen_baselines(self, result_dict):
+        """Generate baselines based on result and dump them in file.
+
+        :param result_dict -  Obtained results, a dict.
+        """
+        prev_baselines = json.load(open(self._baselines_file))
+        for cpu in prev_baselines["hosts"]["instances"]["m5d.metal"]["cpus"]:
+            if cpu["model"] == get_cpu_model_name():
+                baselines = cpu["baselines"]
+        if not baselines:
+            raise Exception("Could not find previous baselines for this CPU!")
+
+        for result in result_dict["results"]:
+            kernel, dist, vm_cfg, workload, *_rest = result.split("/")
+            measurements_node = prev_baselines["measurements"]
+            for measure in measurements_node:
+                for stats in measurements_node[measure]["statistics"]:
+                    if "name" in stats:
+                        fun = stats["name"]
+                    else:
+                        fun = stats["function"]
+                    if measure in result_dict["results"][result]\
+                            and measure in baselines:
+                        result_node = result_dict["results"][result]
+                        function_node = result_node[measure][fun]
+                        value = round(function_node["value"])
+
+                        kernel_node = baselines[measure][kernel]
+                        vm_cfg_node = kernel_node[dist][vm_cfg]
+
+                        # Some tests do not have a workload node (i.e snapshot
+                        # performance.
+                        if fun in vm_cfg_node:
```

Comment:
Adding 
```py
if fun not in vm_cfg_node:
    continue
```

Should get rid of the `pylint: disable=R1702` exception at the top.